demos/form-control/Slider.php

```diff
@@ -0,0 +1,215 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Atk4\Ui\Form\Control;
+
+use Atk4\Ui\View;
+use Atk4\Ui\Js\JsExpression;
+use Atk4\Ui\Js\JsFunction;
+use Atk4\Ui\Js\JsReload;
+
+class Slider extends Input
+{
+    public string $inputType = 'hidden';
+
+    /** The lowest value the slider can be. */
+    public int $min = 0;
+
+    /** The max value the slider can be. */
+    public int $max = 10;
+
+    /** @var float|null The slider step. Set to 0 to disable step. */
+    public $step = 0;
+
+    /** The value the slider will start at. */
+    public int $start = 0;
+
+    /** @var int|null The second value to set in case of a range slider. */
+    public $end = null;
+
+    /** @var int|false Makes sure that the two thumbs of a range slider always need to have a difference of the given value. */
+    public $minRange = false;
+
+    /**  @var int|false Makes sure that the two thumbs of a range slider don't exceed a difference of the given value. */
+    public $maxRange = false;
+
+    /** @var 'number'|'letter' The type of label to display for a labeled slider. Can be number or letter. */
+    public $labelType = 'number';
+
+    /** @var array|null An array of label values which restrict the displayed labels to only those which are defined. */
+    public $restrictedLabels = null;
+
+    /** Whether a tooltip should be shown to the thumb(s) on hover. Will contain the current slider value. */
+    public bool $showThumbTooltip = false;
+    
+    /**
+     * Tooltip configuration used when showThumbTooltip is true
+     * Refer to Tooltip Variations for possible values.
+     *
+     * @var array|null
+     */
+    public $tooltipConfig = null;
+
+    /**
+     * Show ticks on a labeled slider.
+     *'always'will always show the ticks for all labels (even if not shown)
+     * true will display the ticks only if the related label is also shown.
+     *
+     * @var 'always'|bool
+     */
+    public $showLabelTicks = false;
+
+    /** Define smoothness when the slider is moving. */
+    public bool $smooth = false;
+
+    /** Whether labels should auto adjust on window resize. */
+    public bool $autoAdjustLabels = true;
+
+    /** The distance between labels. */
+    public int $labelDistance = 100;
```

Comment:
When is this helpful? Does this affect CSS only?

Reply:
Yes, but it also defines what labels will be shown, depending on space available.